crates/ruff_linter/src/rules/flake8_gettext/mod.rs

```diff
@@ -1,17 +1,50 @@
 //! Rules from [flake8-gettext](https://pypi.org/project/flake8-gettext/).
+use crate::checkers::ast::Checker;
+use crate::preview::is_extended_i18n_function_matching_enabled;
 use ruff_python_ast::name::Name;
 use ruff_python_ast::{self as ast, Expr};
+use ruff_python_semantic::Modules;
 
 pub(crate) mod rules;
 pub mod settings;
 
 /// Returns true if the [`Expr`] is an internationalization function call.
-pub(crate) fn is_gettext_func_call(func: &Expr, functions_names: &[Name]) -> bool {
-    if let Expr::Name(ast::ExprName { id, .. }) = func {
-        functions_names.contains(id)
-    } else {
-        false
+pub(crate) fn is_gettext_func_call(
+    checker: &Checker,
+    func: &Expr,
+    functions_names: &[Name],
+) -> bool {
+    if func
+        .as_name_expr()
+        .map(ast::ExprName::id)
+        .is_some_and(|id| functions_names.contains(id))
+    {
+        return true;
```

Comment:
```suggestion
    if let Some(name) = func.as_name_expr() {
        return functions_names.contains(name.id);
```

I think we can return here even if it's `Some` and _not_ `functions_names.contains(id)`.